From aba590e2b167a1ac3f7abcc3373d834fd4ca53e8 Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:43:43 +0800
Subject: [PATCH 16/23] GSI Android 17 compatibility:
 0016-Flattened-apexes-no-longer-exist-we-ll-deal-with-it-.patch

Change-Id: Iff227ee4aaa2b71b4a66d044891a35461e97c389
---
 init/property_service.cpp | 22 +---------------------
 1 file changed, 1 insertion(+), 21 deletions(-)

diff --git a/init/property_service.cpp b/init/property_service.cpp
index 067e046..6bff088 100644
--- a/init/property_service.cpp
+++ b/init/property_service.cpp
@@ -704,26 +704,6 @@ uint32_t InitPropertySet(const std::string& name, const std::string& value) {
 static Result<void> load_properties_from_file(const char*, const char*,
                                               std::map<std::string, std::string>*);
 
-static bool kernel_supports_capex() {
-    //Put a threshold at >= 5.0
-    struct utsname buf;
-    uname(&buf);
-    const char *where = buf.release;
-    int a = atoi(where);
-    if(a >= 5) return true;
-
-    // If there are vendor apexes, we most likely actually need them
-    auto dir = std::unique_ptr<DIR, decltype(&closedir)>{opendir("/vendor/apex"), closedir};
-    if (!dir) {
-        return false;
-    }
-    for (struct dirent* ent = readdir(dir.get()); ent; ent = readdir(dir.get())) {
-        if(strstr(ent->d_name, "apex")) return true;
-    }
-
-    return false;
-}
-
 /*
  * Filter is used to decide which properties to load: NULL loads all keys,
  * "ro.foo.*" is a prefix match, and "ro.foo.bar" is an exact match.
@@ -827,7 +807,7 @@ static void LoadProperties(char* data, const char* filter, const char* filename,
                 const char *new_value = (*expanded_value).c_str();
 
                 if(strcmp("ro.apex.updatable", key) == 0) {
-                    new_value = kernel_supports_capex() ? "true" : "false";
+                    new_value = "true";
                 }
                 if (it == properties->end()) {
                    (*properties)[key] = new_value;
